aws-iot-thinggroup/src/main/java/software/amazon/iot/thinggroup/Translator.java

```diff
@@ -53,67 +59,31 @@
  * mapping exceptions to appropriate Cloudformation exceptions
  */
 public class Translator {
-    static ProgressEvent<ResourceModel, CallbackContext> translateExceptionToProgressEvent(
-            ResourceModel model, Exception e, Logger logger) {
 
-        HandlerErrorCode errorCode = translateExceptionToErrorCode(e, logger);
-        ProgressEvent<ResourceModel, CallbackContext> progressEvent =
-                ProgressEvent.<ResourceModel, CallbackContext>builder()
-                        .resourceModel(model)
-                        .status(OperationStatus.FAILED)
-                        .errorCode(errorCode)
-                        .build();
-        if (errorCode != HandlerErrorCode.InternalFailure) {
-            progressEvent.setMessage(e.getMessage());
-        }
-        return progressEvent;
-    }
-
-
-    static HandlerErrorCode translateExceptionToErrorCode(Exception e, Logger logger) {
-        logger.log(String.format("Translating exception \"%s\", stack trace: %s",
-                e.getMessage(), ExceptionUtils.getStackTrace(e)));
-
-        // We're handling all the exceptions documented in API docs
-        // https://docs.aws.amazon.com/iot/latest/apireference/API_CreateThingGroup.html#API_CreateThingGroup_Errors
-        // (+same pages for other APIs)
-        // For Throttling and InternalFailure, we want CloudFormation to retry, and it will do so based on the error code.
-        // Reference with Retryable/Terminal in comments for each: https://tinyurl.com/y378qdno
+    static BaseHandlerException translateIotExceptionToHandlerException(
+            String resourceIdentifier, String operationName, IotException e) {
```

Comment:
would a switch statement be better here instead of all these ifs?

Reply:
+1

Reply:
I don't think java natively supports pattern matching of objects in switch statement - apart from  required: 'char, byte, short, int, Character, Byte, Short, Integer, String, or an enum'

Reply:
that's true, sure